### ISSUE TITLE
ManyToOne & JoinColumn Annotation leads to persisting failure in MongoDB

### ISSUE THREAD

**Issue by pdirksen** [Impetus/kundera#512]
MongoDB Version: 2.2.7
Kundera: kundera-mongo 2.9

``` java

@Entity
public class XObs extends Observation {

    @Basic(optional = false)
    @Column(nullable = false, unique = true)
    private String name;

}

@MappedSuperclass
public class Observation implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String id;

    @ManyToOne(optional = false)
    @JoinColumn(nullable = false)
    private Cause cause;

}

@Entity
public class Cause {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String id;

    @Basic(optional = false)
    @Column(nullable = false, unique = true)
    private String name;

}

```

Kundera won't throw an exception, but MongoDB will:

```
[conn25]  gendb3.StructuralObservation Assertion failure !e.eoo() d:\slave\windows_64bit_2008r2+_v2.2\mongo\src\mongo\db\../bson/bsonobjbuilder.h 131
[conn24] end connection 127.0.0.1:7939 (4 connections now open)
[conn25] mongod.exe    ...\src\mongo\util\stacktrace.cpp(182)                           mongo::printStackTrace+0x3e
[conn25] mongod.exe    ...\src\mongo\util\assert_util.cpp(109)                          mongo::verifyFailed+0xdc
[conn25] mongod.exe    ...\src\mongo\bson\bsonobjbuilder.h(131)                         mongo::BSONObjBuilder::appendAs+0x3a
[conn25] mongod.exe    ...\src\mongo\db\indexkey.cpp(380)                            mongo::KeyGeneratorV1::_getKeys+0x398
[conn25] mongod.exe    ...\src\mongo\db\indexkey.cpp(283)                            mongo::KeyGeneratorV1::getKeys+0xde
[conn25] mongod.exe    ...\src\mongo\db\indexkey.cpp(412)                            mongo::IndexSpec::getKeys+0x89
[conn25] mongod.exe    ...\src\mongo\db\index_update.cpp(92)                            mongo::fetchIndexInserters+0xc3
[conn25] mongod.exe    ...\src\mongo\db\index_update.cpp(137)                           mongo::indexRecordUsingTwoSteps+0x229
[conn25] mongod.exe    ...\src\mongo\db\pdfile.cpp(1546)                            mongo::DataFileMgr::insert+0x845
[conn25] mongod.exe    ...\src\mongo\db\pdfile.cpp(1273)                            mongo::DataFileMgr::insertWithObjMod+0x48
[conn25] mongod.exe    ...\src\mongo\db\instance.cpp(754)                            mongo::checkAndInsert+0xb7
[conn25] mongod.exe    ...\src\mongo\db\instance.cpp(814)                            mongo::receivedInsert+0x790
[conn25] mongod.exe    ...\src\mongo\db\instance.cpp(434)                            mongo::assembleResponse+0x607
[conn25] mongod.exe    ...\src\mongo\db\db.cpp(193)                            mongo::MyMessageHandler::process+0xf5
[conn25] mongod.exe    ...\src\mongo\util\net\message_server_port.cpp(86)               mongo::pms::threadRun+0x59a
[conn25] mongod.exe   ..\src\third_party\boost\libs\thread\src\win32\thread.cpp(180)  boost::`anonymous namespace'::thread_start_function+0x21
[conn25] mongod.exe    f:\dd\vctools\crt_bld\self_64_amd64\crt\src\threadex.c(314)      _callthreadstartex+0x17
[conn25] mongod.exe    f:\dd\vctools\crt_bld\self_64_amd64\crt\src\threadex.c(292)      _threadstartex+0x7f
[conn25] kernel32.dll                           BaseThreadInitThunk+0xd
[conn25] insert gendb3.StructuralObservation keyUpdates:0 exception: assertion d:\slave\windows_64bit_2008r2+_v2.2\mongo\src\mongo\db\../bson/bsonobjbuilder.h:131 locks(micros) w:942827 943ms
```

**Comment by mevivs** [comment]
I think XObs is missing @id field?

Could you please elaborate more?

-Vivek

**Comment by kkmishra** [comment]
I have tried following and working fine for me.

Entities:

```
@Entity
public class XObs extends Observation
{
    @Basic(optional = false)
    @Column(nullable = false, unique = true)
    private String name;
}
```

```
@MappedSuperclass
public class Observation
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String id;

    @ManyToOne(optional = false, cascade = CascadeType.ALL)
    @JoinColumn(nullable = false, name = "cause_id")
    private Cause cause;
}
```

```
@Entity
public class Cause
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String id;

    @Basic(optional = false)
    @Column(nullable = false, unique = true)
    private String name;
}
```

CRUD:

```
 @Test
    public void test()
    {

        Cause cause = new Cause();
        cause.setName("new cause");

        XObs xObs = new XObs();
        xObs.setName("xObs1");
        xObs.setCause(cause);

        em.persist(xObs);

        em.clear();

        XObs foundXObs = em.find(XObs.class, xObs.getId());

        Assert.assertNotNull(foundXObs);
        Assert.assertNotNull(foundXObs.getId());
        Assert.assertNotNull(foundXObs.getName());
        Assert.assertNotNull(foundXObs.getCause());
        Assert.assertNotNull(foundXObs.getCause().getId());
        Assert.assertNotNull(foundXObs.getCause().getName());
    }
```

you have not provided join column name in your entity, so it will go as a blank string for column name in database mongodb.

Like:

```
> db.XObs.find()
{ "_id" : "52e132a3e4b0da98558c02e1", "name" : "xObs1", "" : "52e132a3e4b0da98558c02e2" }

```

Ideally it should be field name ie cause, instead of blank string, it will be fixed along with issue #478 and #208 

It will be fixed in 2.10 release

Thanks
Kuldeep
